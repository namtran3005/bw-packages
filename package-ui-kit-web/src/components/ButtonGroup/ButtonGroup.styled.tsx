import * as React from 'react';
import styled from 'styled-components';

interface RootProps {
  fullWidth?: boolean;
  align?: React.CSSProperties['textAlign'];
  style?: React.CSSProperties;
}

const RootComponent: React.FunctionComponent<RootProps> = ({
  /* eslint-disable-next-line @typescript-eslint/no-unused-vars */
  fullWidth,
  /* eslint-disable-next-line @typescript-eslint/no-unused-vars */
  align,
  ...cleanProps
}) => <div {...cleanProps} />;

const Root = styled(RootComponent)`
  display: ${(props) => (props.fullWidth ? 'flex' : 'block')};
  text-align: ${(props) => (props.align ? props.align : 'initial')};

  & > * {
    flex: 1;
    margin-right: 8px !important;

    &:last-child {
      margin-right: 0 !important;
    }
  }
`;

Root.displayName = 'ButtonGroup';
RootComponent.displayName = 'ButtonGroupComponent';

export { Root, RootProps, RootComponent };
