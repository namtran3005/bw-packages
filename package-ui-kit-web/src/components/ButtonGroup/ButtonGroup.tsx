import * as React from 'react';

import { Root, RootProps as ButtonGroupProps } from './ButtonGroup.styled';

const ButtonGroup: React.FunctionComponent<ButtonGroupProps> = (props) => (
  <Root {...props} />
);

export { ButtonGroup, ButtonGroupProps };
