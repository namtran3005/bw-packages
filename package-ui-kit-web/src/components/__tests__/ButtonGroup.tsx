import * as React from 'react';
import { shallow } from 'enzyme';
import { ButtonGroup } from '../ButtonGroup';

const defaultProps = {
  fullWidth: true,
};

describe('ButtonGroup', () => {
  it('should render full width button group', () => {
    const wrapper = shallow(<ButtonGroup {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
