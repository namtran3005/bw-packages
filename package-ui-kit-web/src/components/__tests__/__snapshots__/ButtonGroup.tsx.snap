// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup should render full width button group 1`] = `
<ButtonGroup
  fullWidth={true}
/>
`;
